File: LaboratoryWork/Semester_3/Laboratory_2/GoLang/tasks/avlBalance.go
package tasks

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"

	"laba2/ds" // замените на путь к пакету ds с BST
)

func AVLBalance() {
	fmt.Println("Введите элементы (заканчивается 0):")

	reader := bufio.NewReader(os.Stdin)
	line, _ := reader.ReadString('\n')
	line = strings.TrimSpace(line)
	elems := strings.Split(line, " ")

	tree := &ds.BST{}
	for _, s := range elems {
		val, err := strconv.Atoi(s)
		if err != nil {
			continue
		}
		if val == 0 {
			break
		}
		tree.Root = tree.Insert(tree.Root, val)
	}

	isBalanced := true
	tree.CheckBalance(tree.Root, &isBalanced)

	if isBalanced {
		fmt.Println("YES")
	} else {
		fmt.Println("NO")
	}
}

File: LaboratoryWork/Semester_3/Laboratory_2/C++/src/tasks/AVLBalance.cpp
#include "../../include/AVLBalance.hpp"

using namespace std;

void AVLBalance() {
    cout << "Введите элементы: ";

    BST tree;
    int currentElm;
    cin >> currentElm;
    while (currentElm != 0) {
        tree.root = tree.insert(tree.root, currentElm);
        cin >> currentElm;
    }

    bool isBalanced = true;
    tree.checkBalance(tree.root, isBalanced);
    
    cout << (isBalanced ? "YES" : "NO") << endl;
}